### ISSUE TITLE
Cursor position is incorrect on Linux

### ISSUE THREAD

**Issue by plfiorini** [rburchell/literm_old#14]
![screenshot from 2017-04-18 19-44-52](https://cloud.githubusercontent.com/assets/843710/25144966/faffbfbe-246f-11e7-9e88-38e5956b2c23.png)

konsole shows the cursor one character away from the prompt.

stdout capture:

[log_cursor.txt](https://github.com/rburchell/literm/files/930593/log_cursor.txt)

**Comment by rburchell** [comment]
I think this might be down to font metrics. It doesn't manifest on mac.

**Comment by rburchell** [comment]
The math on the cursor position appears fine. My prompt wants to put the cursor at position 27. Yet visually, it is not where I expect it to be. Hmm.

![image](https://cloud.githubusercontent.com/assets/125863/25147532/1ee905d6-2478-11e7-8ce8-f7e16da1b88a.png)

This seems to improve, but not completely fix it. It looks like there might be some accumulating rounding error somewhere.

```
diff --git a/qml/Yat/Screen.qml b/qml/Yat/Screen.qml
index abebf78..d23cbd1 100644
--- a/qml/Yat/Screen.qml
+++ b/qml/Yat/Screen.qml
@@ -30,10 +30,11 @@ Yat.TerminalScreen {

     property string title: screen.title
     property font font
-    property real fontWidth: fontMetricText.paintedWidth
-    property real fontHeight: fontMetricText.paintedHeight
+    property real fontWidth: fontMetrics.averageCharacterWidth
+    property real fontHeight: fontMetrics.height

     font.family: screen.platformName != "cocoa" ? "monospace" : "menlo"
     anchors.fill: parent
     focus: true

@@ -69,12 +70,9 @@ Yat.TerminalScreen {
         screen.sendKey(event.text, event.key, event.modifiers);
     }

-    Text {
-        id: fontMetricText
-        text: "B"
+    FontMetrics {
+        id: fontMetrics
         font: parent.font
-        visible: false
-        textFormat: Text.PlainText
     }

     Rectangle {
```

**Comment by rburchell** [comment]
Now fixed, by 2e1b00d2a46216ffb2f1017689a24d03b10bcbf3 and 6ce0e5fea7d082a5e5c217235f4af2865ed7ca25.
